Sources/SpeziOnboarding/OnboardingFlow/OnboardingStepIdentifier.swift

```diff
@@ -13,38 +13,103 @@ import SwiftUI
+        if let identifiable = element.view as? any OnboardingIdentifiable {
             let id = identifiable.id
-            hasher.combine(id)
-        } else if let identifiable = view as? any OnboardingIdentifiable {
+            self.identifierKind = .identifiable(id)
+        } else if let identifiable = element.view as? any Identifiable {
             let id = identifiable.id
-            hasher.combine(id)
+            self.identifierKind = .identifiable(id)
         } else {
-            hasher.combine(String(describing: type(of: view as Any)))
+            self.identifierKind = .viewTypeAndSourceLoc
         }
-        self.identifierHash = hasher.finalize()
     }
 
     /// Initializes an identifier using a view type.
     /// - Parameters:
     ///   - onboardingStepType: The class of the view used to initialize the identifier.
     ///   - custom: A flag indicating whether the step is custom.
-    init(onboardingStepType: any View.Type, custom: Bool = false) {
-        self.custom = custom
-        var hasher = Hasher()
-        hasher.combine(String(describing: onboardingStepType))
-        self.identifierHash = hasher.finalize()
+    init(onboardingStepType viewType: (some View).Type, isCustom: Bool = false) {
+        self.isCustom = isCustom
+        self.viewType = viewType
+        self.flowElementSourceLocation = nil
+        self.identifierKind = .viewTypeAndSourceLoc
+    }
+}
+
+
+extension OnboardingStepIdentifier: Hashable {
+    static func == (lhs: Self, rhs: Self) -> Bool {
+        switch (lhs.identifierKind, rhs.identifierKind) {
+        case (.viewTypeAndSourceLoc, .viewTypeAndSourceLoc):
+            lhs.viewType == rhs.viewType && lhs.flowElementSourceLocation == rhs.flowElementSourceLocation
+        case (.identifiable(let lhsValue), .identifiable(let rhsValue)):
+            lhsValue.isEqual(rhsValue)
+        case (.viewTypeAndSourceLoc, .identifiable), (.identifiable, .viewTypeAndSourceLoc):
+            false
+        }
+    }
+    
+    func hash(into hasher: inout Hasher) {
+        switch self.identifierKind {
+        case .viewTypeAndSourceLoc:
+            hasher.combine(ObjectIdentifier(viewType))
+            if let flowElementSourceLocation {
+                hasher.combine(flowElementSourceLocation)
+            }
+        case .identifiable(let value):
+            hasher.combine(ObjectIdentifier(type(of: value)))
+            hasher.combine(value)
+        }
+    }
+}
+
+
+extension OnboardingStepIdentifier: CustomDebugStringConvertible {
+    var debugDescription: String {
+        var desc = "\(Self.self)(isCustom: \(isCustom), viewType: \(viewType), identifierKind: \(identifierKind)"
+        if let sourceLoc = flowElementSourceLocation {
+            desc += ", sourceLoc: \(sourceLoc.fileId);\(sourceLoc.line);\(sourceLoc.column)"
+        }
+        desc += ")"
+        return desc
+    }
+}
+
+
+extension Equatable {
+    func isEqual(_ other: Any) -> Bool {
```

Comment:
Didn't check all usages, but `Any` seems a bit too much here. Can't we use `Hashable`?